### PR TITLE
[log-shipper] Add buffer settings for vector config

### LINKED ISSUE #1352: [log-shipper] Enable buffer
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

The buffer was previously disabled for vector instances. We need this to accumulate logs if the sink is unavailable.
https://github.com/deckhouse/deckhouse/blob/3402a22193ca830e00a6350f200aebd0d07798d2/modules/460-log-shipper/hooks/internal/vector/destination.go#L293-L298

### Proposed Solution

Research the cause of the problem and resolve the issue.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by alex123012**
## Description

This PR adds buffer settings for `ClusterLogDestination` CR
## Why do we need it, and what problem does it solve?

Fix #1352
## What is the expected result?

Users can manage buffer settings for vector in `ClusterLogDestination` crd
## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: log-shipper
type: feature
summary: Add buffer settings for vector config
impact_level: default
```

### CODE REVIEW

File: modules/460-log-shipper/apis/v1alpha1/cluster_log_destination.go [hunk 1]
```diff
@@ -154,3 +156,74 @@ type SplunkSpec struct {
...
+	// Relevant when: type = "memory"
+	Memory BufferMemory `json:"memory,omitempty"`
+
+	// Event handling behavior when a buffer is full.
+	WhenFull BufferWhenFull `json:"whenFull,omitempty"`
+}
+
+type BufferType string
+
+func (t BufferType) ToVectorValue() string {
```
[the remark this answers is not in the document]
Comment (alex123012) [reply modules/460-log-shipper/apis/v1alpha1/cluster_log_destination.go]: done

File: modules/460-log-shipper/crds/cluster-log-destination.yaml [hunk 2]
```diff
@@ -73,7 +73,7 @@ spec:
                     splunk: {}
                     type:
                       enum:
-                      - Splunk
+                        - Splunk
```
[the remark this answers is not in the document]
Comment (alex123012) [reply modules/460-log-shipper/crds/cluster-log-destination.yaml]: done

File: modules/460-log-shipper/crds/cluster-log-destination.yaml [hunk 3]
```diff
@@ -552,3 +552,52 @@ spec:
...
+                        - whenFull
+                  properities:
+                    type:
+                      type: string
+                      description: The type of buffer to use.
+                      enum: ["Disk", "Memory"]
+                    disk:
+                      type: object
+                      properities:
+                        maxSizeBytes:
```
Comment (nabokihms) [inline modules/460-log-shipper/crds/cluster-log-destination.yaml:584]:
```suggestion
                        maxSizeBytes:
```
This is not `Bytes` anymore. 
1. Just maxSize?
2. Examples are required and descriptions should be rewritten.

File: modules/460-log-shipper/hooks/testdata/multiple-dest/manifests.yaml [hunk 4]
```diff
@@ -109,3 +119,8 @@ spec:
       password: c2VjcmV0
   extraLabels:
     foo: bar
+  buffer:
+    type: Disk
+    disk:
+      maxSizeBytes: 268435488
```
Comment (nabokihms) [inline modules/460-log-shipper/hooks/testdata/multiple-dest/manifests.yaml:125]: Test with a string value is required, e.g. 2Gi, 2Mi.

**Review by nabokihms — CHANGES_REQUESTED** [review]

**Review by nabokihms — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

**Pull request merged**